src/components/messenger/list/index.tsx

```diff
@@ -171,34 +195,66 @@ export class Container extends React.Component<Properties, State> {
     );
   }
 
+  stringifyZero() {
+    const stringValue = this.props.zero.padStart(19, '0');
+    const whole = stringValue.slice(0, -18);
+    const decimal = stringValue.slice(-18).slice(0, 4).replace(/0+$/, '');
+    const decimalString = decimal.length > 0 ? `.${decimal}` : '';
+    return `${whole}${decimalString}`;
+  }
+
+  renderRewardsBar() {
```

Comment:
Wonder if it's worth pulling this into a sub-component?

Reply:
not sure if it's worth right now. I will make a note though, if any more changes, we can move this to a sub-component